### PR TITLE
Remove legacy NTRU and OQS

### PR DESCRIPTION
**Pull request by anhu**

### CODE REVIEW

**Review by dgarske — APPROVED** [review]
Looks great. I am glad you caught `WOLFSSL_FILETYPE_RAW` as well.

**Comment by anhu** [comment]
retest this please

**Review by dgarske — APPROVED** [review]

File: examples/server/server.c [hunk 1]
```diff
@@ -1576,10 +1565,6 @@ THREAD_RETURN WOLFSSL_THREAD server_test(void* args)
                 usePskPlus = 1;
                 break;
 
-            case 'n' :
```
Comment (dgarske) [inline examples/server/server.c:1579]: Please also remove `n` from `mygetopt_long` call above. It is now, available.

**Review by dgarske — CHANGES_REQUESTED** [review]
If you are making anymore changes to this PR, one cleanup I spotted in server.c with the `n` arg.

File: examples/server/server.c (hunk 1 above)
Comment (anhu) [reply examples/server/server.c:1579]: Will do.  My goal is to get this diff to 10k lines!

**Review by dgarske — APPROVED** [review]
All looks great. @SparkiDev would you mind also confirming these changes? Thanks

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ standard operating environments as well because of its royalty-free pricing
 and excellent cross platform support. wolfSSL supports industry standards up
 to the current [TLS 1.3](https://www.wolfssl.com/tls13) and DTLS 1.2, is up to
 20 times smaller than OpenSSL, and offers progressive ciphers such as ChaCha20,
-Curve25519, NTRU, and Blake2b. User benchmarking and feedback reports
+Curve25519, Blake2b, OQS TLS 1.3 groups. User benchmarking and feedback reports
```
Comment (SparkiDev) [inline README.md:15]: 'and OQS TLS 1.3 groups'

File: src/tls.c [hunk 3]
```diff
@@ -11021,98 +10200,15 @@ int TLSX_PopulateExtensions(WOLFSSL* ssl, byte isServer)
 #if defined(WOLFSSL_TLS13) && (defined(HAVE_SESSION_TICKET) || !defined(NO_PSK))
     int usingPSK = 0;
 #endif
-#if (defined(HAVE_SUPPORTED_CURVES) && defined(WOLFSSL_TLS13)) || \
-                                                               defined(HAVE_QSH)
+#if (defined(HAVE_SUPPORTED_CURVES) && defined(WOLFSSL_TLS13))
```
Comment (SparkiDev) [inline src/tls.c:10203]: Remove redundant brackets.

File: examples/echoserver/echoserver.c [hunk 4]
```diff
@@ -147,8 +147,7 @@ THREAD_RETURN CYASSL_THREAD echoserver_test(void* args)
 #if defined(CYASSL_DTLS)
     method  = CyaDTLSv1_2_server_method();
 #elif !defined(NO_TLS)
-    #if (defined(WOLFSSL_TLS13) && defined(WOLFSSL_SNIFFER)) || \
-        defined(HAVE_NTRU)
+    #if (defined(WOLFSSL_TLS13) && defined(WOLFSSL_SNIFFER))
```
Comment (SparkiDev) [inline examples/echoserver/echoserver.c:150]: Remove redundant brackets.

**Review by SparkiDev — CHANGES_REQUESTED** [review]
There are two files in certs: ntru-cert.pem and ntru-key.raw.
I believe they can be removed too.

File: src/tls.c (hunk 3 above)
Comment (anhu) [reply src/tls.c:10203]: Cool!  I can completely remove it!

File: README.md (hunk 2 above)
Comment (anhu) [reply README.md:15]: fixed....see next commit.

File: examples/echoserver/echoserver.c (hunk 4 above)
Comment (anhu) [reply examples/echoserver/echoserver.c:150]: fixed.  See next commit.

**Comment by anhu** [comment]
> There are two files in certs: ntru-cert.pem and ntru-key.raw.
> I believe they can be removed too.

Huh...I wonder how I missed those!

**Review by SparkiDev — APPROVED** [review]

**Comment by peterwillcn** [comment]
https://github.com/openwrt/openwrt/commit/7d92bb0509615550b98e2dc71091073c8258d564
```
-- Build files have been written to: /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243
touch /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.built
touch /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.built_check
MAKEFLAGS="" /home/openwrt/master/openwrt/staging_dir/host/bin/ninja  -j1 -C /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243
ninja: Entering directory `/home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243'
[7/8] Linking C executable ustream-example-client
FAILED: ustream-example-client
: && /home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/bin/x86_64-openwrt-linux-musl-gcc -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243=ustream-ssl-2020-12-10-68d09243 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/home/openwrt/master/openwrt/staging_dir/target-x86_64_musl/usr/include/wolfssl -DNDEBUG -L/home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/usr/lib -L/home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/lib -znow -zrelro CMakeFiles/ustream-example-client.dir/ustream-example-client.c.o -o ustream-example-client  libustream-ssl.so  -lubox  -lwolfssl  -lm && :
/home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/11.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /home/openwrt/master/openwrt/staging_dir/target-x86_64_musl/usr/lib/libwolfssl.so: undefined reference to `wc_Sha256Transform'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
make[3]: *** [Makefile:73: /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.built] Error 1
make[3]: Leaving directory '/home/openwrt/master/openwrt/package/libs/ustream-ssl'
time: package/libs/ustream-ssl/wolfssl/compile#1.41#0.49#1.77
    ERROR: package/libs/ustream-ssl failed to build (build variant: wolfssl).
make[2]: *** [package/Makefile:119: package/libs/ustream-ssl/compile] Error 1
make[2]: Leaving directory '/home/openwrt/master/openwrt'
make[1]: *** [package/Makefile:110: /home/openwrt/master/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/openwrt/master/openwrt'
make: *** [/home/openwrt/master/openwrt/include/toplevel.mk:230: world] Error 2
```

**Comment by dgarske** [comment]
> [openwrt/openwrt@7d92bb0](https://github.com/openwrt/openwrt/commit/7d92bb0509615550b98e2dc71091073c8258d564)
> 
> ```
> -- Build files have been written to: /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243
> touch /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.configured_68b329da9893e34099c7d8ad5cb9c940
> rm -f /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.built
> touch /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.built_check
> MAKEFLAGS="" /home/openwrt/master/openwrt/staging_dir/host/bin/ninja  -j1 -C /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243
> ninja: Entering directory `/home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243'
> [7/8] Linking C executable ustream-example-client
> FAILED: ustream-example-client
> : && /home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/bin/x86_64-openwrt-linux-musl-gcc -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243=ustream-ssl-2020-12-10-68d09243 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/home/openwrt/master/openwrt/staging_dir/target-x86_64_musl/usr/include/wolfssl -DNDEBUG -L/home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/usr/lib -L/home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/lib -znow -zrelro CMakeFiles/ustream-example-client.dir/ustream-example-client.c.o -o ustream-example-client  libustream-ssl.so  -lubox  -lwolfssl  -lm && :
> /home/openwrt/master/openwrt/staging_dir/toolchain-x86_64_gcc-11.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/11.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /home/openwrt/master/openwrt/staging_dir/target-x86_64_musl/usr/lib/libwolfssl.so: undefined reference to `wc_Sha256Transform'
> collect2: error: ld returned 1 exit status
> ninja: build stopped: subcommand failed.
> make[3]: *** [Makefile:73: /home/openwrt/master/openwrt/build_dir/target-x86_64_musl/ustream-ssl-wolfssl/ustream-ssl-2020-12-10-68d09243/.built] Error 1
> make[3]: Leaving directory '/home/openwrt/master/openwrt/package/libs/ustream-ssl'
> time: package/libs/ustream-ssl/wolfssl/compile#1.41#0.49#1.77
>     ERROR: package/libs/ustream-ssl failed to build (build variant: wolfssl).
> make[2]: *** [package/Makefile:119: package/libs/ustream-ssl/compile] Error 1
> make[2]: Leaving directory '/home/openwrt/master/openwrt'
> make[1]: *** [package/Makefile:110: /home/openwrt/master/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
> make[1]: Leaving directory '/home/openwrt/master/openwrt'
> make: *** [/home/openwrt/master/openwrt/include/toplevel.mk:230: world] Error 2
> ```

Hi @peterwillcn , the issue does not appear related to this PR. Can you provide your build settings for wolfSSL? Typically this error happens when `--enable-intelasm` or `--enable-armasm` is used and either the platform, compiler or build script isn't including the required assembly files.

Also it seems like PR https://github.com/wolfSSL/wolfssl/pull/4414 resolves this? @JacobBarthelmeh ?

Thanks,
David Garske, wolfSSL

**Comment by peterwillcn** [comment]
@dgarske Perhaps the wolfSSL Release 4.8.2 downstream update should fix this,Thank you  :-)

**Pull request merged**